### ISSUE TITLE
Allow empty options passed to dateFormatter

### ISSUE THREAD

**Issue by dpolivy** [globalizejs/globalize#466]
Repro:
1. Create Globalize object for a locale.
2. Initialize `var options = {};`
3. Call `globalize.dateFormatter(options)`.

Actual:
A exception is thrown:
`[object Object]: E_INVALID_PAR_VALUE: Invalid`{"name":"options","value":{}}`value (undefined).`

Expected:
Globalize should treat an empty options object the same as if no options object were provided.

Some background:
I'm working on a set of handlebars helpers for globalize that allow passing options from the template. Those options come across as an object, regardless of whether they are present. For code simplicity, I just pass this along to `dateFormatter` -- but that doesn't work so well.

NB: I haven't tested this with other formatters, but they may also suffer from the same problem.

**Comment by rxaviers** [comment]
It makes sense. This only happens with `dateFormatter` and `dateParser`, because the default is set using `options = options || { skeleton: "yMd" }`. Perhaps a better implementation would be to leave options as `options = options || {}` and to implement the default at https://github.com/jquery/globalize/blob/master/src/date/expand-pattern.js#L93 recursively.

**Comment by rxaviers** [comment]
cc @jquery/globalize for input.

**Comment by jzaefferer** [comment]
I agree that not options argument and an empty object options argument should be treated the same.

> and to implement the default at https://github.com/jquery/globalize/blob/master/src/date/expand-pattern.js#L93 recursively.

I don't understand that part.

**Comment by rxaviers** [comment]
> I don't understand that part.

```js
default:
    result = expandPattern({ skeleton: "yMd" });
```

**Comment by jzaefferer** [comment]
Gotcha. That seems good.

**Comment by rxaviers** [comment]
This is fixed on the upcoming 1.3.0

```js
Globalize.formatDate(new Date(), {})
'5/22/2017'
```
